clang/lib/Driver/Driver.cpp

```diff
@@ -4144,8 +4144,9 @@ class OffloadingActionBuilder final {
         }
       }
 
-      SmallString<128> LibLoc(TC->getDriver().Dir);
-      llvm::sys::path::append(LibLoc, "/../lib");
+      SmallVector<SmallString<128>, 4> LibLocCandidates;
+      LibLocCandidates.emplace_back(TC->getDriver().Dir + "/../lib");
+      LibLocCandidates.emplace_back(TC->getDriver().SysRoot + "/lib");
```

Comment:
According to my understanding `SysRoot` default value matches `Dir/../`, so probably we can simplify the code and use only one candidate. Right?

Reply:
Hi, @bader 
I just checked that if we don't specify "--sysroot=..." in command line, the "TC->getDriver().SysRoot" will be empty string.
So, we have to keep 2 candidates here.
Thanks very much.

Reply:
Okay. I just noticed that similar logic for HIP/ROCM is implemented in toolchain specific file.
See `RocmInstallationDetector` in https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/ToolChains/AMDGPU.cpp#L171 .
@mdtoguchi, should we move this code to https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/ToolChains/SYCL.cpp?

Reply:
Sure - maybe add a dir vector that holds the known locations in which the libs can be searched for.

Reply:
Hi, @bader and @mdtoguchi 
So, do you mean we need to add a "SyclInstallationDectector" in SYCL.cpp and use it to record search paths for libs, bins, include?
Thanks very much.

Reply:
@jinge90, I don't think we need to add a full installation detector as everything is provided with the compiler.  I think just add the lib search directories when the SYCLToolChain is created.

Reply:
Hi, @mdtoguchi 
I suggest to add a very simple "SyclInstallationDectector" which only provide APIs to query lib search path for now. If we have new request, we can extend the simple "SyclInstallDetector", is it OK?
Thanks very much.

Reply:
Hi, @mdtoguchi  and @bader 
I added a very simple SYCLInstallationDetector, there is no much to do in this function since we don't have many requirements for detecting installation structure currently. If we have some requirements in the future such as adding some option like "-fsycl-installation=...", it is easy to extend this class.
Thanks very much.

Reply:
> I think just add the lib search directories when the SYCLToolChain is created.

I thought that Mike suggestion is that we move `addSYCLDeviceLibs` to SYCLToolChain.